helix-core/src/main/java/org/apache/helix/util/HelixUtil.java

```diff
@@ -177,6 +178,39 @@ public static String serializeByComma(List<String> objects) {
       String metadataStoreAddress, ClusterConfig clusterConfig,
       List<InstanceConfig> instanceConfigs, List<String> liveInstances,
       List<IdealState> idealStates, List<ResourceConfig> resourceConfigs) {
+    return getIdealAssignmentForWagedFullAutoImpl(metadataStoreAddress, clusterConfig,
+        instanceConfigs, liveInstances, idealStates, resourceConfigs, true);
+  }
+
+  /**
+   * Returns the expected ideal ResourceAssignments for the given resources in the cluster
+   * calculated using the read-only WAGED rebalancer. The returned result is based on partition
+   * state mapping.
+   * @param metadataStoreAddress
+   * @param clusterConfig
+   * @param instanceConfigs
+   * @param liveInstances
+   * @param idealStates
+   * @param resourceConfigs
+   * @return
+   */
+  public static Map<String, ResourceAssignment> getIdealPartitionMapForWagedFullAuto(
```

Comment:
The names are not clear enough, IMO.
How about getTargetAssignmentForWagedFullAuto and getImmediateAssignmentForWagedFullAutoImpl ?

Reply:
Sure, I was also thinking what's a good way to differentiate these two.